src/test/java/com/exactpro/th2/conn/TestEvent.java

```diff
@@ -0,0 +1,155 @@
+            return (SubscriberMonitor) () -> {
+            };
+        }).when(router).subscribeAll(any(), any());
+
+        eventDispatcher = mock(EventDispatcher.class);
+        doAnswer(invocation -> {
+            EventHolder eventHolder = invocation.getArgument(0);
+            EventType eventType = eventHolder.getType();
+
+            eventDispatcher.store(eventHolder.getEvent(),
+                    parentIds.get(eventType) != null ? parentIds.get(eventType) : rootID);
+            return null;
+        }).when(eventDispatcher).store(any());
+
+        doAnswer(invocation -> {
+            event = invocation.getArgument(0);
+            parentId = invocation.getArgument(1);
+            return null;
+        }).when(eventDispatcher).store(any(), any());
+
+        messageSender = new MessageSender(serviceProxy, router, eventDispatcher,
+                EventID.newBuilder().setId("stubID").build());
+        messageSender.start();
+    }
+
+    @AfterEach
+    public void clear() {
+        event = null;
+        parentId = null;
+    }
+
+    public void sendIncorrectMessage() throws Exception {
+        RawMessageBatch rawMessageBatch = RawMessageBatch.newBuilder()
+                .addMessages(RawMessage.newBuilder().build())
+                .build();
+
+        doThrow(new IllegalStateException("error")).when(serviceProxy).sendRaw(any(), any());
+        messageListener.handler("stubValue", rawMessageBatch);
+    }
+
+    @Test
+    public void eventHasBodyTest() throws Exception {
+        sendIncorrectMessage();
+        event.addSubEvent(Event.start());
+
+        ByteString body = event.toProto(EventUtils.toEventID(parentId)).getBody();
+        Assertions.assertEquals("[{\"data\":\"java.lang.IllegalStateException: error\",\"type\":\"message\"}," +
+                "{\"data\":\"Cannot send message. Message body in base64:\",\"type\":\"message\"},{\"data\":\"\"," +
+                "\"type\":\"message\"},{\"data\":\"java.lang.IllegalStateException: error\",\"type\":\"message\"}]", body.toStringUtf8());
+    }
+
+    @Test
+    public void eventHasNameTest() throws Exception {
+        sendIncorrectMessage();
+        event.addSubEvent(Event.start());
+
+        String name = event.toProto(EventUtils.toEventID(parentId)).getName();
+        Assertions.assertEquals("Raw message sending error", name);
+    }
+
+    @Test
+    public void sentMessageWithParentEventIDTest() throws Exception {
+        RawMessageBatch rawMessageBatch = RawMessageBatch.newBuilder()
+                .addMessages(RawMessage.newBuilder()
+                        .setParentEventId(EventID.newBuilder()
+                                .setId("RawMessageParentEventID")).build())
+                .build();
+
+        doThrow(new IllegalStateException("error")).when(serviceProxy).sendRaw(any(), any());
+        messageListener.handler("stubValue", rawMessageBatch);
+
+        event.addSubEvent(Event.start());
```

Comment:
What's this for?

Reply:
events must contain at least 2 instances to set parentEventID

Reply:
<img width="848" alt="image" src="https://user-images.githubusercontent.com/86191842/160138099-d0f7f396-6a8a-44f2-84a5-65cf8e522d06.png">

Reply:
> at least 2 instances

Code above will also work with zero

Reply:
yes, but we need to store our event under test